pkg/virt-config/feature-gates.go

```diff
@@ -24,19 +24,20 @@ package virtconfig
 */
 
 const (
-	CPUManager            = "CPUManager"
-	IgnitionGate          = "ExperimentalIgnitionSupport"
-	LiveMigrationGate     = "LiveMigration"
-	CPUNodeDiscoveryGate  = "CPUNodeDiscovery"
-	HypervStrictCheckGate = "HypervStrictCheck"
-	SidecarGate           = "Sidecar"
-	GPUGate               = "GPU"
-	HostDevicesGate       = "HostDevices"
-	SnapshotGate          = "Snapshot"
-	HotplugVolumesGate    = "HotplugVolumes"
-	HostDiskGate          = "HostDisk"
-	VirtIOFSGate          = "ExperimentalVirtiofsSupport"
-	MacvtapGate           = "Macvtap"
+	CPUManager             = "CPUManager"
+	IgnitionGate           = "ExperimentalIgnitionSupport"
+	LiveMigrationGate      = "LiveMigration"
+	SRIOVLiveMigrationGate = "SRIOVLiveMigration"
```

Comment:
Please document the meaning (and the downsides) of your new feature somewhere.

Reply:
Done

Reply:
Correct me if I am wrong but currently there is no documentation that explain how a feature-gate affects the system. 
If one would like to know more then that a feature is enabled he would need to investigate the code.

I think those kind of comments should be placed right next to the deceleration (it will also appear on the IDE when you hover on it), 
until we will have a document about feature-gate with an explanation of what a feature-gate does (if there is a doc that does please let me know).

When this feature-gate will be redundant we could revert this PR (or at lease follow what it change) so this comment will also be removed

Reply:
> There is documentation that handles each feature independently, e.g. https://kubevirt.io/2020/Live-migration.html
> 

I meant to a document that describes all the feature-gates we have with explanation for what they do and their side-effects
(e.g SRIOVLiveMigration side-effect is running virt-launcher with CAP_SYS_RESOURCE)

> In code we express detailed behavior using tests.
> It is fine to document what it does here, but it should not include "how" as that is an implementation detail that can change. And the how (e.g. adding the capability) can go away without removing the gating (which has other reasons to exists, like the missing devices during the migration period).
>
> So my suggestion is to leave only the first line (`// SRIOVLiveMigrationGate enable's Live Migration for VM's with SRIOV interfaces.`)
> 

In general I agree with you.
I think that SRIOVLiveMigration feature-gate is a special case because it controls when virt-launcher pod runs with extra privileges (CAP_SYS_RESOURE).
Which degrade its security and might increase attack surface.

I was thinking about leaving the first line like you suggest 
and adding a note about the capability, something like:
`When enabled virt-launcher runs with CAP_SYS_RESOURE`
Or 
`When enabled virt-launcher runs with extra privileges`
What you think?

Reply:
I think you should document this in the place the action is taking place. Here you declare a flag and someone else is going to use it... the addition of the capability is there, not here.

So if you want be pedantic here, add the comment about this in the place the action is taken place, ref to a BZ and mention this is due that and can be removed when it is resolved. Finally you could also add this in the release-notes by adding it in the proper section in the PR description.

Reply:
> For the last commit: I think you should add to `Live Migration for VM's with SRIOV interfaces should be blocked.` something like "even though the feature-gate is open".

Done